embark-ui/src/components/TransactionDecoder.js

```diff
@@ -20,20 +20,26 @@ class TransactionDecoder extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      transactionHash: this.props.transactionHash || ''
+      transactionHash: this.props.transactionHash || '',
+      isRawTxHash: false
     };
   }
 
   handleTransactionHashChange(event) {
     const transactionHash = event.target.value;
-    this.setState({transactionHash});
+    this.setState({...this.state, transactionHash});
+  }
+
+  toggleTransactionHashType(event) {
+    const isRawTxHash = event.target.checked;
+    this.setState({...this.state, isRawTxHash});
   }
 
   fetchTransaction(e) {
     e.preventDefault();
     if (this.state.transactionHash !== '') {
       this.props.history.push({
-        search: `hash=${this.state.transactionHash}`
+        search: `hash=${this.state.transactionHash}&isRawTxHash=${this.state.isRawTxHash}`
```

Comment:
The idea is that we're reusing the existing transaction decoder and simply tell it that the hash in question is actually a raw transaction hash

Reply:
Why do we have the transaction decoder not decoding transaction?
I would have think that we pass only raw tx to the decoder.

If we have a transactionHash we can use the search?